src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -374,9 +374,10 @@ internal override bool HasPointerType
                 if (_lazyType != null)
                 {
                     Debug.Assert(_lazyType.Value.DefaultType.IsPointerType() ==
-                        IsPointerFieldSyntactically());
+                        IsPointerFieldSyntactically() ||
+                        _lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer);
```

Comment:
>_lazyType.Value.DefaultType.Kind == SymbolKind.FunctionPointer [](start = 24, length = 62)

The purpose of the assert was to make sure result ```sPointerFieldSyntactically``` is consistent with result of binding. I think the change leaves this purpose unserved.